qt/python/mantidqt/project/projectsaver.py

```diff
@@ -71,13 +71,13 @@ def save_project(self, file_name, workspace_to_save=None, plots_to_save=None, in
         writer.write_out()
 
     @staticmethod
-    def _return_interfaces_dicts(file_name, interfaces_to_save):
+    def _return_interfaces_dicts(directory, interfaces_to_save):
         interfaces = []
         for interface, encoder in interfaces_to_save:
             # Add to the dictionary encoded data with the key as the first tag in the list on the encoder attributes
             try:
                 tag = encoder.tags[0]
-                encoded_dict = encoder.encode(interface, file_name)
+                encoded_dict = encoder.encode(interface, directory)
```

Comment:
Quick question. Why do the encoders need to know the directory where their information will be saved?

Reply:
Specifically Instrument View needs to know so it can save the mask information there and load it back

It is also conceivable that other things may require to save their own data so this was added as a requirement to encoders.

Reply:
Okay. I can see that the mask saving is quite complicated so having it saved to the project file when we have a separate file format anyway is reasonable. We should try and limit those that save out to separate files in future though.